spec/unit/indirector/catalog/compiler_spec.rb

```diff
@@ -257,14 +257,14 @@
     def a_legacy_request_that_contains(facts, format = :pson)
       request = Puppet::Indirector::Request.new(:catalog, :find, "hostname", nil)
       request.options[:facts_format] = format.to_s
-      request.options[:facts] = Puppet::Util.uri_encode(facts.render(format))
+      request.options[:facts] = Puppet::Util.uri_query_encode(facts.render(format))
```

Comment:
None of the specs in compiler_spec route through `fact_handler` as far as I can tell, so doing `uri_query_encode` here (as we also do on line 260 of this file) is intended to simulate that I believe. This also updates a previous spec that was using `uri_encode` instead of using our new `uri_query_encode`.